bedrock/base/templates/macros.html

```diff
@@ -301,6 +301,28 @@ <h3>{{ _('Your download link was sent.') }}</h3>
   </section>
 {%- endmacro %}
 
+{% macro fxa_link_fragment(entrypoint, action='signin', utm_params={}) -%}
+  {% set fxa_path = '' %}
+  {% set fxa_queries = [
+    'service=sync',
+    'context=fx_desktop_v3'
+  ] %}
+
+  {% if action == 'signup' %}
+    {% set fxa_path = 'signup' %}
+  {% else %}
+    {% do fxa_queries.insert(0, 'action=email') %}
+  {% endif %}
+  {% do fxa_queries.append('entrypoint={}'.format(entrypoint)) %}
+  {% for param, val in utm_params.items() %}
+    {% if val and param in ['campaign', 'content', 'medium', 'source'] %}
+      {% do fxa_queries.append('utm_{}={}'.format(param, val.replace('/', '%2F'))) %}
```

Comment:
Would it be worthwhile URL encoding the values here, rather than special-casing for  `/`?

Reply:
We can address this in a follow up PR.